cmd/plume/create.go

```diff
@@ -0,0 +1,136 @@
+
+		Name:        "create-instances",
+		Summary:     "Create cluster on GCE",
+		Usage:       "-image <gce image name> -n <number of instances>",
+		Description: "Upload os image to Google Storage bucket and create image in GCE",
+		Flags:       *flag.NewFlagSet("create", flag.ExitOnError),
+		Run:         runCreate,
+	}
+	createProject      string
+	createZone         string
+	createMachine      string
+	createBaseName     string
+	createImageName    string
+	createConfig       string
+	createNumInstances int
+)
+
+func init() {
+	cmdCreate.Flags.StringVar(&createProject, "project", "coreos-gce-testing", "found in developers console")
+	cmdCreate.Flags.StringVar(&createZone, "zone", "us-central1-a", "gce zone")
+	cmdCreate.Flags.StringVar(&createMachine, "machine", "n1-standard-1", "gce machine type")
+	cmdCreate.Flags.StringVar(&createBaseName, "name", "", "prefix for instances")
+	cmdCreate.Flags.StringVar(&createImageName, "image", "", "GCE image name")
+	cmdCreate.Flags.StringVar(&createConfig, "config", "", "path to cloud config file")
+	cmdCreate.Flags.IntVar(&createNumInstances, "n", 1, "number of instances")
+	cli.Register(cmdCreate)
+}
+
+func runCreate(args []string) int {
+	if len(args) != 0 {
+		fmt.Fprintf(os.Stderr, "Unrecognized args in plume create-instances: %v\n", args)
+		return 2
+	}
+
+	if createImageName == "" {
+		fmt.Fprintf(os.Stderr, "Must specifcy GCE image with '-image' flag\n")
+		return 2
+	}
+	// if base name is unspecified use image name
+	if createBaseName == "" {
+		createBaseName = createImageName
+	}
+
+	var cloudConfig string
+	if createConfig != "" {
+		b, err := ioutil.ReadFile(createConfig)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Could not read cloud config file: %v\n", err)
+			return 1
+		}
+		cloudConfig = string(b)
+	}
+
+	client, err := auth.GoogleClient()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Authentication failed: %v\n", err)
+		return 1
+	}
+
+	var vms []*gceVM
+	for i := 0; i < createNumInstances; i++ {
+		name, err := nextName(client, createProject, createZone, createBaseName)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed creating name for vm: %v\n", err)
+			return 1
+		}
+		vm, err := createVM(client, createProject, createZone, createMachine, name, createImageName, cloudConfig)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed creating vm: %v\n", err)
+			return 1
+		}
+		vms = append(vms, vm)
+		fmt.Printf("Instance %v created\n", name)
+	}
+	fmt.Printf("All instances created, add your ssh keys here: https://console.developers.google.com/project/%v/compute/metadata/sshKeys\n", createProject)
+	for _, vm := range vms {
+		fmt.Printf("To access %v use cmd:\n", vm.name)
+		fmt.Printf("ssh -o UserKnownHostsFile=/dev/null -o CheckHostIP=no -o StrictHostKeyChecking=no core@%v\n", vm.extIP)
+	}
+
+	return 0
+}
+
+// nextName returns the next available numbered name or the given base name.
+// snipped from: https://github.com/golang/build/blob/master/cmd/gomote/list.go
```

Comment:
Yes, add the same go author copyright to the top. Also keep the existing
comments/links. That is useful to know.
On Apr 8, 2015 10:35 PM, "Patrick Baxter" <notifications@github.com> wrote:

> In cmd/plume/create.go
> <https://github.com/coreos/mantle/pull/38#discussion_r28036188>:
>
> > +			return 1
> > +		}
> > +		vms = append(vms, vm)
> > +		fmt.Printf("Instance %v created\n", name)
> > +	}
> > +	fmt.Printf("All instances created, add your ssh keys here: https://console.developers.google.com/project/%v/compute/metadata/sshKeys\n", createProject)
> > +	for _, vm := range vms {
> > +		fmt.Printf("To access %v use cmd:\n", vm.name)
> > +		fmt.Printf("ssh -o UserKnownHostsFile=/dev/null -o CheckHostIP=no -o StrictHostKeyChecking=no core@%v\n", vm.extIP)
> > +	}
> > +
> > +	return 0
> > +}
> > +
> > +// nextName returns the next available numbered name or the given base name.
> > +// snipped from: https://github.com/golang/build/blob/master/cmd/gomote/list.go
>
> So just add the Go author copyright up with ours? Should I still mention
> which functions have the code and a link to where it is from?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/mantle/pull/38/files#r28036188>.
>